core/src/main/kotlin/org/evomaster/core/problem/rest/service/RestSampler.kt

```diff
@@ -190,18 +201,176 @@ class RestSampler : Sampler<RestIndividual>() {
         val actions = mutableListOf<RestAction>()
         val n = randomness.nextInt(1, config.maxTestSize)
 
+        usedObjects.clearLists()
         (0 until n).forEach {
             actions.add(sampleRandomAction(0.05))
         }
+        val objInd = RestIndividual(actions, SampleType.RANDOM, mutableListOf(), usedObjects.copy())
+        usedObjects.clearLists()
+        return objInd
+    }
+
+    private fun proposeObject(g: Gene): Pair<ObjectGene, Pair<String, String>> {
+        var restrictedModels = mutableMapOf<String, ObjectGene>()
+        when (g::class) {
+            ObjectGene::class -> {
+                // If the gene is an object, select a suitable one from the Model CLuster (based on the Swagger)
+                restrictedModels = modelCluster.filter{ model ->
+                    (model.value as ObjectGene).fields
+                            .map{ it.name }
+                            .toSet().containsAll((g as ObjectGene).fields.map { it.name }) }.toMutableMap()
+                if (restrictedModels.isEmpty()) return Pair(ObjectGene("none", mutableListOf()), Pair("none", "Not_found"))
+                val ret = randomness.choose(restrictedModels)
+                return Pair(ret, Pair(ret.name, "Complete_object"))
+            }
+            else -> {
+                modelCluster.forEach { k, model ->
+                    val fields = model.fields.filter { field ->
+                        when (field::class) {
+                            OptionalGene::class -> g::class === (field as OptionalGene).gene::class
+                            else -> g::class === field::class
+                        }
+                    }
+                    restrictedModels[k] = ObjectGene(model.name, fields)
+                }
+            }
+        }
 
-        return RestIndividual(actions, SampleType.RANDOM)
+        //Having filtered the objects, build the probability map.
+        val likely = likelyhoodsExtended(g.getVariableName(), restrictedModels)
+                .toList()
+                .sortedBy { (_, value) -> -value}
```

Comment:
i don't understand. what's the point to sorting it if then u collect the results in an unsorted map?